File: benchmarks/hash_stress.py
n = 5000
m = {}
seed = 99991
for _ in range(n):
    seed = (seed * 1664525 + 1013904223) % 4294967296
    k = seed % n
    m[k] = k * 2
total = 0
for _ in range(n):
    seed = (seed * 1664525 + 1013904223) % 4294967296
    k = seed % n
    v = m.get(k)
    if v is not None: total += v
print(total)

File: benchmarks/cpp/hash_stress.cpp
#include <cstdio>
#include <unordered_map>
int main() {
    const int n = 5000;
    std::unordered_map<long long,long long> m;
    long long seed = 99991, total = 0;
    for (int i = 0; i < n; i++) {
        seed = (seed*1664525+1013904223)%4294967296LL;
        long long k = seed % n;
        m[k] = k * 2;
    }
    for (int i = 0; i < n; i++) {
        seed = (seed*1664525+1013904223)%4294967296LL;
        long long k = seed % n;
        auto it = m.find(k);
        if (it != m.end()) total += it->second;
    }
    printf("%lld\n", total);
}
